src/runtime/runtime_stm32.go

```diff
@@ -4,12 +4,55 @@ package runtime
 
 import (
 	"device/arm"
+	"device/stm32"
+	"machine"
 )
 
 type timeUnit int64
 
 const tickMicros = 1 // TODO
 
+const (
+	// Flash Access Control Register flag values.
+	FLASH_ACR_LATENCY_0 = 0x00000004
+	FLASH_ACR_LATENCY_1 = 0x00000002
+	FLASH_ACR_LATENCY_2 = 0x00000004
+	FLASH_ACR_HLFCYA    = 0x00000008
+	FLASH_ACR_PRFTBE    = 0x00000010
+	FLASH_ACR_PRFTBS    = 0x00000020
+
+	// Reset and Clock Control Control Register flag values.
+	RCC_CFGR_SW_HSI = 0
+	RCC_CFGR_SW_HSE = 1
+	RCC_CFGR_SW_PLL = 2
+
+	RCC_CFGR_SWS_HSI = 0x00000000
+	RCC_CFGR_SWS_HSE = 0x00000004
+	RCC_CFGR_SWS_PLL = 0x00000008
+
+	RCC_CFGR_PPRE1_DIV_NONE = 0x00000000
+	RCC_CFGR_PPRE1_DIV_2    = 0x00000400
+	RCC_CFGR_PPRE1_DIV_4    = 0x00000500
+	RCC_CFGR_PPRE1_DIV_8    = 0x00000600
+	RCC_CFGR_PPRE1_DIV_16   = 0x00000700
+
+	RCC_CFGR_PLLMUL_2  = 0x00000000
+	RCC_CFGR_PLLMUL_3  = 0x00040000
+	RCC_CFGR_PLLMUL_4  = 0x00080000
+	RCC_CFGR_PLLMUL_5  = 0x000C0000
+	RCC_CFGR_PLLMUL_6  = 0x00100000
+	RCC_CFGR_PLLMUL_7  = 0x00140000
+	RCC_CFGR_PLLMUL_8  = 0x00180000
+	RCC_CFGR_PLLMUL_9  = 0x001C0000
+	RCC_CFGR_PLLMUL_10 = 0x00200000
+	RCC_CFGR_PLLMUL_11 = 0x00240000
+	RCC_CFGR_PLLMUL_12 = 0x00280000
+	RCC_CFGR_PLLMUL_13 = 0x002C0000
+	RCC_CFGR_PLLMUL_14 = 0x00300000
+	RCC_CFGR_PLLMUL_15 = 0x00340000
+	RCC_CFGR_PLLMUL_16 = 0x00380000
+)
```

Comment:
It doesn't seem to me like these constants belong in the runtime or machine package. It's very unfortunate that the .svd files provided by STMicro do not contain these bitfields, but I don't think putting them directly in the machine and runtime packages isn't the right thing to do either. Maybe they can be moved to src/device/stm32/stm32f103xx-bitfields.go, for example?

Reply:
There is src/device/nrf, src/device/arm, and src/device/avr. I think it's fine to add another directory.
Git doesn't track directories, only files. A file that is already tracked is not ignored by `.gitignore`.

Reply:
I could use the `-f` flag to force the commit of such a bitfields file to the correct directory `src/device/stm32/stm32f103xx-bitfields.go` which seems like would accommodate this use case.

Reply:
I think I did it before simply by commenting out the lines in the .gitignore file before adding and uncommenting again afterwards. But maybe you can also try adding an exception to the .gitignore file.